x-pack/plugins/security_solution/public/cloud_security_posture/components/csp_details/insights_tab_csp.tsx

```diff
@@ -5,19 +5,119 @@
 import { MisconfigurationFindingsDetailsTable } from './misconfiguration_findings_details_table';
+import { VulnerabilitiesFindingsDetailsTable } from './vulnerabilities_findings_details_table';
 
 /**
  * Insights view displayed in the document details expandable flyout left section
  */
+
+interface CspFlyoutPanelProps extends FlyoutPanelProps {
+  params: {
+    path: PanelPath;
+    hasMisconfigurationFindings: boolean;
+    hasVulnerabilitiesFindings: boolean;
+  };
+}
+interface CspFlyoutPanels extends FlyoutPanels {
+  left: CspFlyoutPanelProps;
+}
+
+// Type guard to check if the path is a PathPanel
+const isPathPanel = (path: PanelPath): path is PanelPath => {
+  return typeof path?.tab === 'string';
+};
+
 export const InsightsTabCsp = memo(
   ({ name, fieldName }: { name: string; fieldName: 'host.name' | 'user.name' }) => {
+    /* Using type cast here because originally I thought paths are part of Params when its actually its own field. The correct
+    way to solve this is by putting path outside params for every single openLeftPanel call in this folder, but given the time frame
+    We decided to do this for now */
+    const panels = useExpandableFlyoutState() as CspFlyoutPanels;
+
+    const getDefaultTab = () =>
+      panels.left?.params?.hasMisconfigurationFindings
+        ? CspInsightLeftPanelSubTab.MISCONFIGURATIONS
+        : panels.left?.params?.hasVulnerabilitiesFindings
+        ? CspInsightLeftPanelSubTab.VULNERABILITIES
+        : 'undefined';
+
+    const [activeInsightsId, setActiveInsightsId] = useState(
+      panels.left?.params?.path?.subTab || getDefaultTab()
+    );
+
+    const insightsButtonsNew: EuiButtonGroupOptionProps[] = useMemo(() => {
+      const buttons: EuiButtonGroupOptionProps[] = [];
+
+      if (panels.left?.params?.hasMisconfigurationFindings) {
+        buttons.push({
+          id: CspInsightLeftPanelSubTab.MISCONFIGURATIONS,
+          label: (
+            <FormattedMessage
+              id="xpack.securitySolution.flyout.left.insights.misconfigurationButtonLabel"
+              defaultMessage="Misconfiguration"
+            />
+          ),
+          'data-test-subj': 'misconfigurationTabDataTestId',
+        });
+      }
+
+      if (panels.left?.params?.hasVulnerabilitiesFindings) {
+        buttons.push({
+          id: CspInsightLeftPanelSubTab.VULNERABILITIES,
+          label: (
+            <FormattedMessage
+              id="xpack.securitySolution.flyout.left.insights.vulnerabilitiesButtonLabel"
+              defaultMessage="Vulnerabilities"
+            />
+          ),
+          'data-test-subj': 'vulnerabilitiesTabDataTestId',
+        });
+      }
+      return buttons;
+    }, [
+      panels.left?.params?.hasMisconfigurationFindings,
+      panels.left?.params?.hasVulnerabilitiesFindings,
+    ]);
+
+    const onTabChange = (id: string) => {
+      setActiveInsightsId(id);
+    };
+
+    if (!isPathPanel(panels.left?.params?.path)) return <></>;
```

Comment:
this condition does not seem straightforward for me to understand, could you add a comment explaining why the early return when there is no path panel?

Or I'm wondering if that could be replaced with a logic that checks for the `insightsButtons` length as it is directly related to the code, i.e.

```ts
    if (insightsButtons.length === 0) {
      return null;
    }
```

Reply:
originally I thought, when user open left panel `path` should always be provided, but now that u showed me the above, your solution looks better and more straightforward to understand